### PR TITLE
Refactor proc macros, add more knobs

### PR DESCRIPTION
**Pull request by vorot93**
This PR contains general refactoring of `#[tokio::main]` and `#[tokio::test]` macros along with additional `core_threads` and `max_threads` knobs.

Note: `core_threads` knob sets the runtime to `threaded` or `basic` depending on the value. Current `threaded_scheduler` and `basic_scheduler` knobs are only retained for compatibility with existing code and are overridden if `core_threads` is set.

### CODE REVIEW

File: tokio/Cargo.toml [hunk 1]
```diff
@@ -54,7 +54,7 @@ io-driver = ["mio", "lazy_static"]
 io-util = ["memchr"]
 # stdin, stdout, stderr
 io-std = ["rt-core"]
-macros = ["tokio-macros"]
+macros = ["rt-threaded", "tokio-macros"]
```
Comment (LucioFranco) [inline tokio/Cargo.toml]: why is this added?

Comment (vorot93) [reply tokio/Cargo.toml]: remade to require only rt-core
